src/clj/rems/application/approver_bot.clj

```diff
@@ -5,28 +5,19 @@
 
 (def bot-userid "approver-bot")
 
-(defn- generate-commands [app]
-  (when (and (application-util/is-handler? app bot-userid)
-             (= :application.state/submitted (:application/state app))
-             (empty? (:application/blacklist app)))
-    [{:type :application.command/approve
-      :actor bot-userid
-      :time (time/now)
-      :application-id (:application/id app)
-      :comment ""}]))
+(defn- should-approve? [application]
+  (empty? (:application/blacklist application)))
 
-(defn- generate-commands-for-application-id [app-id]
-  (generate-commands (applications/get-unrestricted-application app-id)))
+(defn- generate-commands [event application]
+  (when (= :application.event/submitted (:event/type event)) ;; approver bot only reacts to fresh applications
+    (when (application-util/is-handler? application bot-userid)
+      (when (should-approve? application)
```

Comment:
Would `(when (and` be better than nested `when`s?

Reply:
fair enough :)